packages/service/src/handlers/login-handler.ts

```diff
@@ -26,30 +25,26 @@ export class LoginHandler extends HttpHandler {
       throw new BadRequestHttpError('No Identity or Identity Provider sent with the request')
     }
 
-    let agentUrl: string
-
     const webId = context.authn.webId
 
-    // see if authorization agent exists for that WebID use it
-    let oidcSession = await this.sessionManager.getOidcSession(webId)
+    const oidcSession = await this.sessionManager.getOidcSession(webId)
 
-    if (oidcSession && oidcSession.info.isLoggedIn) {
+    if (oidcSession.info.isLoggedIn) {
       return { status: 204, headers: {} }
-    } else if (oidcSession) {
-      agentUrl = oidcSession.info.clientAppId!
-    } else {
-      agentUrl = uuid2agentUrl(randomUUID())
-      oidcSession = new Session({ storage: this.sessionManager.storage, }, webId);
-      await this.sessionManager.setAgentUrl2WebIdMapping(agentUrl, webId)
     }
 
+    // TODO I think this should aso go into the manager, however I couldn't find
+    //      the right way to go about it
+    const agentUrl = uuid2agentUrl(randomUUID());
+    await this.sessionManager.setAgentUrl2WebIdMapping(agentUrl, webId)
+
```

Comment:
I think you could move it into `SessionManager#getOidcSession` where you added `if (!session)`.
We never want an oidcSession that doesn't have an agent associated with it. Actually we should use `SessionManager#buildSaiSession` which expects `clientId` (so the agentUrl)